src/libstore/unix/build/local-derivation-goal.cc

```diff
@@ -102,7 +102,7 @@ void handleDiffHook(
     }
 }
 
-const Path LocalDerivationGoal::homeDir = "/homeless-shelter";
+const Path LocalDerivationGoal::homeDir = "/proc/homeless-shelter";
```

Comment:
`/proc` doesn't exist on macOS, maybe doesn't exist on any of the *BSDs we have (fledgling?) support for, and probably won't exist on Windows (which has seen some development recently).

Reply:
I don't think that changing the path will cause new trouble - it would just mean that a build script which does `mkdir -p $HOME/.local/cache` (for example) will create `/proc/homeless-shelter` instead of `/homeless-shelter`. But would it help if I add a condition that will set it to `/proc/homeless-shelter` only on Linux?

Reply:
Is a single-user installation allowed to write to /dev?
MacOS has /dev